lib/active_model/serializer.rb

```diff
@@ -68,14 +68,13 @@ def serializer_for(resource, options = {})
             ArraySerializer
           end
         else
-          search_list = build_serializer_class_list(resource, options)
-          result = search_list.map do |klass_name|
-                     Serializer.serializers_cache.fetch_or_store(klass_name) do
-                       _const_get(klass_name)
-                     end
-                   end
-
-          result.find { |serializer| !serializer.nil? }
+          each_possible_serializer(resource, options) do |klass_name|
+            serializer = Serializer.serializers_cache.fetch_or_store(klass_name) do
+              _const_get(klass_name)
+            end
+            return serializer unless serializer.nil?
```

Comment:
Hmm,  I try to avoid return in a block. Break would work,  too, no?

Reply:
Break could work yes. I find return from a block quite idomatic, but feel free to change the style if you feel strongly about it.

The goal here was first and foremost to reduce the overhead.